src/lib/ngsiNotify/Notifier.cpp

```diff
@@ -50,32 +51,312 @@ Notifier::~Notifier (void)
+
+  LM_W(("KZ: In templateNotify"));
+  //
+  // 1. Verb/Method
+  //
+  if (verb == NOVERB)
+  {
+    // Default verb/method is POST
+    verb = POST;
+  }
+  method = verbName(verb);
+
+
+  //
+  // 2. URL
+  //
+  macroSubstitute(&url, httpInfo.url, ce);
+
+
+  //
+  // 3. Payload
+  //
+  if (httpInfo.payload == "")
+  {
+    payload = "{" + ce.toJson(renderFormat, attrsOrder) + "}";
+    LM_W(("KZ: payload: %s", payload.c_str()));
+  }
+  else
+  {
+    macroSubstitute(&payload, httpInfo.payload, ce);
+  }
+
+
+  //
+  // 4. URI Params (Query Strings)
+  //
+  for (std::map<std::string, std::string>::const_iterator it = httpInfo.qs.begin(); it != httpInfo.qs.end(); ++it)
+  {
+    std::string key   = it->first;
+    std::string value = it->second;
+
+    macroSubstitute(&value, it->second, ce);
+    qs[key] = value;
+  }
+
+
+  //
+  // 5. HTTP Headers
+  //
+  for (std::map<std::string, std::string>::const_iterator it = httpInfo.headers.begin(); it != httpInfo.headers.end(); ++it)
+  {
+    std::string key   = it->first;
+    std::string value = it->second;
+
+    macroSubstitute(&value, it->second, ce);
+    headers[key] = value;
+  }
+
+
+  //
+  // 6. Split URI in protocol, host, port and path
+  //
+  std::string  protocol;
+  std::string  host;
+  int          port;
+  std::string  uriPath;
+
+  if (!parseUrl(url, host, port, uriPath, protocol))
+  {
+    LM_E(("Runtime Error (not sending NotifyContextRequest: malformed URL: '%s')", httpInfo.url.c_str()));
+    return false;
+  }
+
+
+  //
+  // 7. Add URI params from template to uriPath
+  //
+  std::string  uri = uriPath;
+
+  if (headers.size() != 0)
+  {
+    uri += "?";
+
+    int ix = 0;
+    for (std::map<std::string, std::string>::iterator it = qs.begin(); it != qs.end(); ++it)
+    {
+      if (ix != 0)
+      {
+        uri += "&";
+      }
+
+      uri += it->first + "=" + it->second;
+      ++ix;
+    }
+  }
+
+
+  //
+  // 8. Send the request
+  //
+  //    NOTE: the HTTP headers are sent to httpRequestSend via parameter 'extraHeaders'
+  //
+  std::string  out;
+  int          r;
+
+  LM_W(("KZ: Sending notification: %s [%s]", uri.c_str(), payload.c_str()));
+  r = httpRequestSend(host,
+                      port,
+                      protocol,
+                      method,
+                      tenant,
+                      ce.entityId.servicePath,
+                      xauthToken,
+                      uri,
+                      "application/json",
+                      payload,
+                      fiwareCorrelator,
+                      renderFormatToString(renderFormat),
+                      true,                // Use Rush if CLI '--rush' allows it
+                      true,                // wait for response
+                      &out,
+                      headers,
+                      "application/json",  // Accept Format
+                      -1);                 // Timeout in milliseconds, depends on CLI '--httpTimeout'
```

Comment:
Actually, the field is named `-httpTimeout`. Note that double '-' is not used in the CLI (except for `--version`).

2 occurrences (this and other in another place).